### PR TITLE
[RHPAM-2008] HTTP 503 error is shown in Monitoring [KIECLOUD-227] OpenShiftStartupStrategy in S2I amq template creates two configMaps and Kie Server is not deployed

### PR DESCRIPTION
**Pull request by rhtevan**
@errantepiphany 
This PR can be reviewed but need to be merged together with the following PR.

Related PRs
https://github.com/jboss-container-images/jboss-kie-modules/pull/251

Related JIRAs
https://issues.jboss.org/projects/RHPAM/issues/RHPAM-2008

Signed-off-by: Evan Zhang <evan.zhang@redhat.com>

Thanks for submitting your Pull Request!

Please make sure your PR meets the following requirements:

- [x] Pull Request title is properly formatted: `[RHPAM-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Attached commits represent units of work and are properly formatted
- [x] You have read and agreed to the Developer Certificate of Origin (DCO) (see `CONTRIBUTING.md`)
- [x] Every commit contains `Signed-off-by: Your Name <yourname@example.com>` - use `git commit -s`

### CODE REVIEW

File: templates/rhpam74-kieserver-externaldb.yaml [hunk 1]
```diff
@@ -676,6 +687,8 @@ objects:
             containerPort: 8888
             protocol: TCP
           env:
+          - name: WORKBENCH_SERVICE_NAME
```
Comment (sutaakar) [inline templates/rhpam74-kieserver-externaldb.yaml]: What should happen when user doesn't set this environment variable, considering that this variable is not mandatory and doesn't have default value?

Comment (rhtevan) [reply templates/rhpam74-kieserver-externaldb.yaml]: @spolti Correct me if I'm wrong.
@sutaakar  If this environment variable is undefined, the hook script will ignore the 'curl' command. In such case, this KIE Server will be treated as unmanaged.

Comment (spolti) [reply templates/rhpam74-kieserver-externaldb.yaml]: correct, the WORKBENCH_CONTROLLER and KIE_SERVER_ID must exists for the hook be triggered.
there there is only the KIE_SERVER_ID the configmap will be updated only, the controller event (curl request) will not be fired.

**Comment by errantepiphany** [comment]
@spolti , @rhtevan : Should this PR no longer be labeled as "on hold"? Directly asking, is it ready to be merged along with https://github.com/jboss-container-images/jboss-kie-modules/pull/251 )? What about the the KieServerOpenShift.java (non-test) file here https://github.com/kiegroup/droolsjbpm-integration/pull/1827/files#diff-56e6143d7ef0fcf12799656bd766b93b ? Is that a pre-requisite to all of this?

**Comment by spolti** [comment]
That's a question for Evan, I am not sure if all PRs are ready, at least the hooks needs to be reviewed again.

**Comment by rhtevan** [comment]
> @spolti , @rhtevan : Should this PR no longer be labeled as "on hold"? Directly asking, is it ready to be merged along with [jboss-container-images/jboss-kie-modules#251](https://github.com/jboss-container-images/jboss-kie-modules/pull/251) )? What about the the KieServerOpenShift.java (non-test) file here https://github.com/kiegroup/droolsjbpm-integration/pull/1827/files#diff-56e6143d7ef0fcf12799656bd766b93b ? Is that a pre-requisite to all of this?

@errantepiphany Yes. Both hook script and Java change are all pre-requisites to this PR.

**Comment by rhtevan** [comment]
@spolti Added the fix for issue (https://issues.jboss.org/browse/KIECLOUD-227) as per this investigation (https://issues.jboss.org/browse/KIECLOUD-227?focusedCommentId=13738515&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-13738515). 

In short, the KIE Server DC is missing kie-server-id label comparing to other KIE-Server only templates.

@spolti @sutaakar @jakubschwan  See if one of you could validate the fix.

**Comment by jakubschwan** [comment]
@rhtevan I take a look on the fix and it works now, thanks! Only one Kie Server config maps was created.

**Comment by rhtevan** [comment]
@errantepiphany It's ready to merge.

**Pull request merged**